### DISCUSSION TITLE
Allow for TextOptions, FontColor, and Point in DrawText

### DISCUSSION THREAD

**Discussion by Bonfire** [SixLabors/ImageSharp.Drawing#220] [Q&A] [answered]
Currently there exists no overloaded method under the `DrawTextExtensions` class that allows the user to specify TextOptions, FontColor, and Point. One such example is if you attempted to draw text (using `.DrawText()`) that is horizontally centered, with a font color of black, at the point (100, 100). To my knowledge, the above example is currently not possible with the existing methods. I believe that the above example is contains very reasonable and basic functionality that I would expect out of a Fonts library.

If this is in error (maybe it can be done using DrawingOptions?), please feel free to close this discussion. Thank you.

EDIT: This may need to be moved to the [ImageSharp.Drawing](https://github.com/SixLabors/ImageSharp.Drawing) repository instead. I am currently not able to transfer the discussion, but you are free to do so. Or, if you'd rather this discussion be closed, I can re-create the discussion at that repo.

**Comment by JimBobSquarePants** [comment, marked as answer]
```c#
public static IImageProcessingContext DrawText(this IImageProcessingContext source, TextOptions textOptions, string text, Color color)
```
https://docs.sixlabors.com/api/ImageSharp.Drawing/SixLabors.ImageSharp.Drawing.Processing.DrawTextExtensions.html#SixLabors_ImageSharp_Drawing_Processing_DrawTextExtensions_DrawText_IImageProcessingContext_TextOptions_System_String_Color_

`TextOptions` contains a property called `Origin` that is used to set the location.
https://docs.sixlabors.com/api/Fonts/SixLabors.Fonts.TextOptions.html#SixLabors_Fonts_TextOptions_Origin

**Reply by Bonfire** [reply]
Thank you for pointing that out! Do you suggest the user re-use `TextOptions` and modify the `Origin` property before each `DrawText()` call? It feels a little awkward having to set the `Origin` outside of the `DrawText()` as opposed to within the `DrawText()`. Unless I am misunderstanding?

Here's how I picture a typical user would approach this using the method you've mentioned above:
```cs
myImage.Mutate(myImage =>
{
    myTextOptions.Origin = new Point(x, y);
    myImage.DrawText(mytextOptions, myString, myFontColor);
});
```

Is that the intended approach? Or is there a way to modify the origin in a more succinct way that I am not seeing?

Thank you for your time.

**Reply by JimBobSquarePants** [reply]
I wouldn’t recommend sharing options at all since they contain so many action specific properties font size, wrapping length, etc. We’re also adding rich text support via those options with text run collections.
